### PR TITLE
add docker resources and file server 

### PR DESCRIPTION
**Pull request by AlexMorreale**
## what / why 
 - Dockerfile for local and production development 
 - docker-compose.yml for local testing
 - serve.go to serve files

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# check for dep binary so we dont run this
+# check file || curl ....
+#RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+
+RUN cd /src/ch \
+#    && dep ensure \
+    && GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o certificate-host
+
+# final stage
+FROM alpine
```
Comment (pinkertonpg) [inline Dockerfile:16]: Do we even need alpine here? Can we do a raw binary, or no due to the file we'll have to drop in there?

Comment (phantasm66) [reply Dockerfile:16]: I think since the image needs to have a filesystem (for the files/dirs added below), then it needs an OS.

**Review by pinkertonpg — APPROVED** [review]
Talked in person about cert stuff

File: Dockerfile (hunk 1 above)
Comment (pinkertonpg) [reply Dockerfile:16]: That makes sense, wasn't sure if we could get around that by somehow packaging it in the binary.

**Review by phantasm66 — APPROVED** [review]

**Comment by samhitov** [comment]
I don't know what this is for.

**Comment by AlexMorreale** [comment]
oh sorry @samhitov  and @mikowitz  we need a way to host a certificate for the dex authenticator.

this is a basic file server in GoLang.

**Review by mikowitz — APPROVED** [review]

**Review by samhitov — APPROVED** [review]
That makes sense, thanks. Mind adding your explanation to the README before merging?

**Pull request merged**